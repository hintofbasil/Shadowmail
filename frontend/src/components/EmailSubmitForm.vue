<template>
  <div>
    <div class="row">
      <div class="offset-by-two eight columns">
        <form
          method="POST"
          id="new-email-form"
          class="input-then-button-layout"
        >
          <input type="submit" value="â†©" v-on:click="submit" />
          <span class="submit-span">
            <input
              v-model="email"
              type="email"
              name="email"
              class="email-input"
              id="new-email-input"
              placeholder="you@domain.com"
            />
          </span>
        </form>
      </div>
    </div>
    <div class="row">
      <div class="offset-by-two eight columns">
        <EmailSubmitSuccess v-if="new_email" v-bind:email="new_email" />
      </div>
    </div>
  </div>
</template>

<script>
import axios from "axios";

import EmailSubmitSuccess from "@/components/EmailSubmitSuccess.vue";

const NEW_EMAIL_URL = "/api/new";

export default {
  components: {
    EmailSubmitSuccess
  },
  data: () => {
    return {
      email: "",
      new_email: ""
    };
  },
  methods: {
    submit: async function(event) {
      event.preventDefault();
      const submitData = {
        email: this.email
      };
      const response = await axios.post(NEW_EMAIL_URL, submitData);
      this.new_email = response.data.email;
    }
  }
};
</script>

<style lang="scss" scoped>
.input-then-button-layout {
  margin-bottom: 2rem;

  > :first-child {
    float: right;
    padding: 0;
    width: 38px; // Height is set in Skeleton
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }

  span {
    display: block;
    overflow: hidden;

    > :first-child {
      width: 100%;
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
      border-right: none;
    }
  }
}
</style>
