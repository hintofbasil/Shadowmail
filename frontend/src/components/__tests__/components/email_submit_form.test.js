import React from 'react';
import renderer from 'react-test-renderer';

import EmailSubmitForm from '../../email_submit_form';

describe('render', () => {
  it('renders successfully', () => {
    const tree = renderer
      .create(
        <EmailSubmitForm />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
