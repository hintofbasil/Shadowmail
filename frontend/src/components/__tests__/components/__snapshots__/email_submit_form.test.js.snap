// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders successfully 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="offset-by-two eight columns"
    >
      <form
        className="input-then-button-layout"
        id="new-email-form"
        method="POST"
      >
        <input
          onClick={[Function]}
          type="submit"
          value="â†©"
        />
        <span
          className="submit-span"
        >
          <input
            className="email-input"
            id="new-email-input"
            name="email"
            onChange={[Function]}
            placeholder="you@domain.com"
            type="email"
            value=""
          />
        </span>
      </form>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="offset-by-two eight columns"
    />
  </div>,
]
`;
